### DISCUSSION TITLE
When use reverse proxy, I can't visit it 4873

### DISCUSSION THREAD

**Discussion by yp910108** [verdaccio/verdaccio#3216] [Q&A] [answered]
version: v5.11.0

I'm sorry, my English is awful. I'm trying to describe my problem. I hope you can help me. Thank you!

I use pm2 start my npm on port 4873 and I use nginx proxy it on 80.

When I visit http://hostname it's fine. But when i visit http://hostname:4873, i'm got error`Refused to connect to 'http://hostname:4873/-/verdaccio/data/packages' because it violates the document's Content Security Policy`.

And i'm restart my npm by `pm2 restart verdaccio`, i can visit http://hostname:4873, but when i visit http://hostname, i'm got the same error.

Here is my Nginx configuration:
```
location / {
    proxy_pass http://127.0.0.1:4873/;
    proxy_set_header Host $host:$server_port;
    proxy_set_header X-Forwarded-For $remote_addr;
    proxy_set_header X-Forwarded-Proto $scheme;
}
```
I hope both http://hostname and http://hostname:4873 can be accessed.

Am I using it wrong? I hope you understand what I mean. Thank you very much!

**Comment by juanpicado** [comment, marked as answer]
No worries, 🤗 I'm not native english speaker, I'll do my best to understand.

Regarding your conf, I'm not an expert with Nginx, but I think the value is `proxy_set_header Host $http_host;`

I've written an article here time ago https://medium.com/verdaccio/setting-up-verdaccio-on-digitalocean-61b5d08e4f0d

On a server, I have this running with pm2, verdaccio and Nginx and this config below works fine for me.

```
location / {		
		proxy_pass http://127.0.0.1:4873/;
                proxy_set_header X-Forwarded-Host $host;
        	proxy_set_header Host $host;
		proxy_set_header X-Forwarded-Proto $scheme;
		proxy_set_header X-Real-IP $remote_addr;
		proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;		
        }
```

Keep on mind, **that verdaccio uses the `host` HTTP header to figure out where you are running the server,** if you don't provide this header (which NGINX does it  using `proxy_set_header`) it will use `localhost`. So you cannot use two domains at the same time, otherwise the browser will throw [CORS errors](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS).

I hope that helps.

**Reply by yp910108** [reply]
I've learned. Only one domain can be used at a time. 😄
Thank you for your dedication，verdaccio helped me a lot.👍
